### PR TITLE
Fix 9399

### PR DESCRIPTION
**Pull request by triump2020**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #9399

## What this PR does / why we need it:
Before this PR :
  If nblk(non-apendable block)'.createTime <= txn.startTS < nblk. deleteTime=nblk.commitTs ,  then txn will lost nblk's deletes.

### CODE REVIEW

**Comment by JackTan25** [comment]
please give more docker-compose tests, thanks

**Comment by triump2020** [comment]
Do not merge it  until pass throug all the test in my local env.

**Review by XuPeng-SH — APPROVED** [review]

**Review by JackTan25 — APPROVED** [review]

File: pkg/vm/engine/disttae/logtailreplay/partition_state.go [hunk 1]
```diff
@@ -400,7 +400,29 @@ func (p *PartitionState) HandleMetadataInsert(ctx context.Context, input *api.Ba
 				}
 				iter.Release()
 			}
-
+			if !entryStateVector[i] && len(entry.DeltaLoc) != 0 {
```
Comment (reusee) [inline pkg/vm/engine/disttae/logtailreplay/partition_state.go:403]: Why not add these conditions to line 380? Seems exactly the same.

**Review by reusee — CHANGES_REQUESTED** [review]

File: pkg/vm/engine/disttae/logtailreplay/partition_state.go (hunk 1 above)
Comment (triump2020) [reply pkg/vm/engine/disttae/logtailreplay/partition_state.go]: Done.

**Comment by triump2020** [comment]
@matrix-meow rerun failed

**Review by nnsgmsone — APPROVED** [review]

**Review by reusee — APPROVED** [review]

**Pull request merged**